chain-abstraction/sol-implementation/edge_challenge_manager.go

```diff
@@ -257,45 +242,28 @@ func (e *specEdge) Bisect(
 	return lower, upper, nil
 }
 
-func (e *specEdge) ConfirmByTimer(ctx context.Context, ancestorIds []protocol.EdgeId) error {
+func (e *specEdge) ConfirmByTimer(ctx context.Context) error {
 	s, err := e.Status(ctx)
 	if err != nil {
 		return err
 	}
 	if s == protocol.EdgeConfirmed {
 		return nil
 	}
-	var assertionHash protocol.AssertionHash
-	if len(ancestorIds) != 0 {
-		topLevelAncestorId := ancestorIds[len(ancestorIds)-1]
-		topLevelAncestor, topLevelErr := e.manager.GetEdge(ctx, topLevelAncestorId)
-		if topLevelErr != nil {
-			return topLevelErr
-		}
-		if topLevelAncestor.IsNone() {
-			return fmt.Errorf("did not find edge with id %#x for specified top level ancestor", topLevelAncestorId)
-		}
-		topEdge := topLevelAncestor.Unwrap()
-		challengeLevel := topEdge.GetChallengeLevel()
-		if !challengeLevel.IsBlockChallengeLevel() {
-			return errors.New("top level ancestor must be a block challenge edge")
-		}
-		assertionHash = protocol.AssertionHash{
-			Hash: common.Hash(topEdge.ClaimId().Unwrap()),
-		}
-	} else {
-		assertionHash = protocol.AssertionHash{
-			Hash: e.inner.ClaimId,
-		}
+	if e.GetChallengeLevel() != protocol.NewBlockChallengeLevel() {
+		return errors.New("only block challenge edges can be confirmed by time")
+	}
+	if e.ClaimId().IsNone() {
+		return errors.New("only root edges can be confirmed by time")
+	}
+	assertionHash := protocol.AssertionHash{
+		Hash: e.inner.ClaimId,
 	}
 	assertionCreation, err := e.manager.assertionChain.ReadAssertionCreationInfo(ctx, assertionHash)
 	if err != nil {
 		return err
 	}
-	ancestors := make([][32]byte, len(ancestorIds))
-	for i, r := range ancestorIds {
-		ancestors[i] = r.Hash
-	}
+	ancestors := make([][32]byte, 0)
```

Comment:
super nit. Can we add a note here as to why are we passing an empty ancestors here? (it is only clear after looking at the solidity code)